io/net/network_test.py

```diff
@@ -0,0 +1,277 @@
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Prudhvi Miryala <mprudhvi@linux.vnet.ibm.com>
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+#
+
+"""
+check the statistics of interface, test big ping
+test lro and gro and interface
+"""
+
+import time
+import hashlib
+import netifaces
+from avocado import main
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils import genio
+
+
+class NetworkTest(Test):
+    '''
+    To test different types of pings
+    '''
+
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        smm = SoftwareManager()
+        pkgs = ["ethtool", "net-tools"]
+        detected_distro = distro.detect()
+        if detected_distro.name == "Ubuntu":
+            pkgs.extend(["openssh-client", "iputils-ping"])
+        elif detected_distro.name == "SuSE":
+            pkgs.extend(["openssh", "iputils"])
+        else:
+            pkgs.extend(["openssh-clients", "iputils"])
+        for pkg in pkgs:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel("%s package is need to test" % pkg)
+        interfaces = netifaces.interfaces()
+        interface = self.params.get("interface")
+        if interface not in interfaces:
+            self.cancel("%s interface is not available" % interface)
+        self.iface = interface
+        self.peer = self.params.get("peer_ip")
+        if not self.peer:
+            self.cancel("No peer provided")
+        if not self.ping_check("-c 2"):
+            self.cancel("No connection to peer")
+        self.mtu = self.params.get("mtu", default=1500)
+
+    def test_mtu_set(self):
+        '''
+        set mtu size
+        '''
+        if not self.set_mtu_peer(self.mtu):
+            self.fail("Failed to set mtu in peer")
+        time.sleep(10)
+        if not self.set_mtu_host(self.mtu):
+            self.fail("Failed to set mtu in host")
+        time.sleep(10)
+
+    def test_gro(self):
+        '''
+        Test GRO
+        '''
+        ro_type = "gro"
+        ro_type_full = "generic-receive-offload"
+        if not self.receive_offload_state(ro_type_full):
+            self.fail("Could not get state of %s" % ro_type)
+        if self.receive_offload_state(ro_type_full) == 'fixed':
+            self.fail("Can not change the state of %s" % ro_type)
+        self.receive_offload_toggle_test(ro_type, ro_type_full)
+
+    def test_lro(self):
+        '''
+        Test LRO
+        '''
+        ro_type = "lro"
+        ro_type_full = "large-receive-offload"
+        if not self.receive_offload_state(ro_type_full):
+            self.fail("Could not get state of %s" % ro_type)
+        if self.receive_offload_state(ro_type_full) == 'fixed':
+            self.fail("Can not change the state of %s" % ro_type)
```

Comment:
why fail if not scenario is not supported.. better to cancel

Reply:
File "/usr/lib/python2.7/site-packages/avocado_framework-68.0-py2.7.egg/avocado/core/test.py", line 1025, in fail
    raise exceptions.TestFail(message)
TestFail: Can not change the state of lro

Reply:
@abdhaleegit that is because, we should cancel in setUp(), and fail in testX().
I can take this as an enhancement.